### DISCUSSION TITLE
How to interact with batch_id of a 3DTile ?

### DISCUSSION THREAD

**Discussion by valentinMachado** [iTowns/itowns#2054] [Q&A]
Some context:

We would like to contribute to itowns with our [project @ud-viz/browser](https://github.com/VCityTeam/UD-Viz/tree/master/packages/browser) and we need to clean our code to be align with the itowns framework. [see](https://github.com/VCityTeam/UD-Viz/issues/584). This is quite a big issue for us and I splitted it in many issues, the first one to be treated is this [one](https://github.com/VCityTeam/UD-Viz/issues/594) 

The real question:

Our class [CityObject](https://github.com/VCityTeam/UD-Viz/blob/master/packages/browser/src/Component/Itowns/3DTiles/Model/CityObject.js) is relevant for itowns ?

> Basically it records an interval in vertex array of a 3D tile (that can be retrieved with a batch id), some props of the batch table (that can be retrieved as well with a batch id) and record 3D tile id

My questions:

I might miss something since Iam not aware of all the feature/use case of this class, but for me it's not relevant since everything in a 3DTileLayer context can be retrieve with a batch id thus this is simply a sugar batch id, do you agree ?
Also we can have some interaction with these "CityObject" to apply some material for example. Where this code should belong ? As some Util method in 3DTileLayer (such as setBatchIDMaterial(batchID, material)) ? As some Util function in our project (such as set3DTileBatchIDMaterial(tile, batchID, material) ?

**Comment by valentinMachado** [comment]
I will identify needs on our side and come back then with features proposition (or not), so I am closing this

**Comment by jailln** [comment]
Thanks for opening this discussion. I agree with you that the CityObject notion should not be present in iTowns. However, this class was implemented in ud-viz to compensate some lacks in the 3DTiles API to fill certain needs (such as changing the style of features identified by their tile id and batch id or by some attribute in the batch table). Therefore I think we should improve itowns 3DTiles API. One way to proceed would be to start from the usecases and especially your usecases. Feel free to reopen with usecases that are not fulfilled by the current API :)
